tools/github-team-user-store/GitHubTeamUserStore/GitHubTeamUserStore/GitHubEventClient.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            _gitHubClient = CreateClientWithGitHubEnvToken(productHeaderName);
+        }
+
+        /// <summary>
+        /// This method creates a GitHubClient using the GITHUB_TOKEN from the environment for authentication
+        /// </summary>
+        /// <param name="productHeaderName">This is used to generate the User Agent string sent with each request. The name used should represent the product, the GitHub Organization, or the GitHub username that's using Octokit.net (in that order of preference).</param>
+        /// <exception cref="ArgumentException">If the product header name is null or empty</exception>
+        /// <exception cref="ApplicationException">If there is no GITHUB_TOKEN in the environment</exception>
+        /// <returns>Authenticated GitHubClient</returns>
+        public virtual GitHubClient CreateClientWithGitHubEnvToken(string productHeaderName)
+        {
+            if (string.IsNullOrEmpty(productHeaderName))
+            {
+                throw new ArgumentException("productHeaderName cannot be null or empty");
+            }
+            var githubToken = Environment.GetEnvironmentVariable("GITHUB_TOKEN");
+            if (string.IsNullOrEmpty(githubToken))
+            {
+                throw new ApplicationException("GITHUB_TOKEN cannot be null or empty");
+            }
+            var gitHubClient = new GitHubClient(new ProductHeaderValue(productHeaderName))
+            {
+                Credentials = new Credentials(githubToken)
+            };
+            return gitHubClient;
+        }
+
+        /// <summary>
+        /// Using the authenticated GitHubClient, call the RateLimit API to get the rate limits.
+        /// </summary>
+        /// <returns>Octokit.MiscellaneousRateLimit which contains the rate limit information.</returns>
+        public async Task<MiscellaneousRateLimit> GetRateLimits()
+        {
+            return await _gitHubClient.RateLimit.GetRateLimits();
+        }
+
+        /// <summary>
+        /// Write the current rate limit and remaining number of transactions.
+        /// </summary>
+        /// <param name="prependMessage">Optional message to prepend to the rate limit message.</param>
+        public async Task WriteRateLimits(string prependMessage = null)
+        {
+            var miscRateLimit = await GetRateLimits();
+            CoreRateLimit = miscRateLimit.Resources.Core.Limit;
+            // Get the Minutes till reset.
+            TimeSpan span = miscRateLimit.Resources.Core.Reset.UtcDateTime.Subtract(DateTime.UtcNow);
+            // In the message, cast TotalMinutes to an int to get a whole number of minutes.
+            string rateLimitMessage = $"Limit={miscRateLimit.Resources.Core.Limit}, Remaining={miscRateLimit.Resources.Core.Remaining}, Limit Reset in {(int)span.TotalMinutes} minutes.";
+            if (prependMessage != null)
+            {
+                rateLimitMessage = $"{prependMessage} {rateLimitMessage}";
```

Comment:
```suggestion
                rateLimitMessage = $"{prependMessage}{rateLimitMessage}";
```

I think that with that space, if there is no `prependMessage`, the message will always be indented by one space, which I guess we don't want.

Reply:
I'm not changing this. If prepend message is null it simply doesn't add a prependMessage to the rateLimitMessage